src/mono/mono/metadata/class-internals.h

```diff
@@ -1600,6 +1600,18 @@ m_method_alloc0 (MonoMethod *method, guint size)
 	return mono_mem_manager_alloc0 (m_method_get_mem_manager (method), size);
 }
 
+/*
+ * mono_mem_manager_get_default:
+ *
+ *   Return the default memory manager. Memory allocated from
+ * this memory manager will never be freed/unloaded.
+ */
+static inline MonoMemoryManager*
+mono_mem_manager_get_default (void)
+{
+	return mono_domain_ambient_memory_manager (mono_get_root_domain ());
```

Comment:
Ambient was intended as a way to mark places where we need to replace the call with the 'proper' MemoryManager, but on places where we know we want the default like this we can just call `mono_domain_memory_manager`. At some point in the future I'll have to audit most/all of the ambient calls and make sure to get the correct MM, or we leak.

Reply:
Trying to get rid of mentions of 'domain' in the runtime code.

Reply:
Right, but we're still planning to have the root domain, right? I guess it makes sense to say we should always just call `mono_mem_manager_get_default` and get rid of `mono_domain_memory_manager` now that the mirror is off, since there's only one domain.

Reply:
We will have the root domain, but I don't think it should be used for much.